### PR TITLE
Add new module contribution guide for extended library modules

### PR DESCRIPTION
**Pull request by NipunaRanasinghe**
## Purpose
$subject.

Fixes https://github.com/ballerina-platform/ballerina-extended-library/issues/506

## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

### CODE REVIEW

**Comment by abeykoon** [comment]
@NipunaRanasinghe - I read through LGTM. However a few points, 

1. Did we decided not to have samples folder? I remember we had issues maintaining those samples making sure they work across version upgrades. 
2.  It will be great if we can mention a specific format for package.md and module.md files 
3. It will be great to link to our keyword system. We have a doc maintained mentioning all keywords. Lib developers better select appropriate keywords from that list rather introduce their own. Maybe we can make it a wiki. https://docs.google.com/spreadsheets/d/1BZtaCARF_mIJiRcvw6jdzIeNEK0dNuwxZlpSjRb68IA/edit#gid=1033521037

**Comment by NipunaRanasinghe** [comment]
@abeykoon Thanks for your suggestions and please find my responses below.

> 1. Did we decided not to have samples folder? I remember we had issues maintaining those samples making sure they work across version upgrades.

IIRC we decided to make it optional, as we cannot generalize this rule for the generated connectors. And I was thinking of enforcing to add all the samples under the `examples/` directory in repo root, to make it consistent with the standard library structure (even though we some differences between standard and extended libraries module structures, IMO we better stick to common conventions wherever possible). WDYT?

> 2. It will be great if we can mention a specific format for package.md and module.md files

Yes that's already in the roadmap. One question I came up while creating templates for the above two files was, whether we can have a common template (i.e. defined in https://docs.google.com/document/d/1WZg1ix9MpKDTQ_OOm3NVqhQ5L0VGNaUQ1LRmjp-SMO4/edit?usp=sharing) for both handwritten and generated connector modules or, whether we need two separate templates. Appreciate your thought on this.

> 3. It will be great to link to our keyword system. We have a doc maintained mentioning all keywords. Lib developers better select appropriate keywords from that list rather introduce their own. Maybe we can make it a wiki. https://docs.google.com/spreadsheets/d/1BZtaCARF_mIJiRcvw6jdzIeNEK0dNuwxZlpSjRb68IA/edit#gid=1033521037

+1 to document this information and mention in the module contribution guide.

**Comment by abeykoon** [comment]
@NipunaRanasinghe 

1. +1 to have a similar structure with std libs repository. 
2. We do not need separate templates. It can be same. Only thing is, generated connectors/triggers might have a sub-set of information.  
3. Yes. We can address in a separate PR. Pls track.

**Review by abeykoon — APPROVED** [review]

**Comment by NipunaRanasinghe** [comment]
@abeykoon I've udpated the guide with the `Package.md` and `Module.md` templates as suggested. I'll create a separate PR on keyword recommendations, as the doc we are having seems to be outdated. 

Appreciate your review on the new changes.

File: docs/module-contribution-guide.md [hunk 1]
```diff
@@ -0,0 +1,332 @@
...
+
+### The `LICENSE` file [Required]
+
+This file contains the license of the module. All the Ballerina Standard Library modules use the `Apache-2` license.
+
+### The `README.md` file [Required]
+
+This file contains the description and the build steps of the module. Usually the content of this file is similar to the `Module.md` and `Package.md` files of the Ballerina module.
+
+It also includes the status badges for the module including the `Build`, `CodeCoverage`, `Trivy`, `GraalVM`, `last commit`, and the `Open Issues` badges.
```
Comment (ThisaruGuruge) [inline docs/module-contribution-guide.md:291]: We don't have all these badges in the extended libraries, right?

Comment (NipunaRanasinghe) [reply docs/module-contribution-guide.md]: Yes we have a separate set of labels. Updated in https://github.com/ballerina-platform/ballerina-extended-library/pull/505/commits/a735a8d72fe43bbbd39ad075d28ba5c396bb2b67

File: docs/module-contribution-guide.md [hunk 2]
```diff
@@ -0,0 +1,332 @@
...
+├── gradle.properties
+├── gradlew
+├── gradlew.bat
+├── settings.gradle
+├── spotbugs-exclude.xml
+```
+
+### The `LICENSE` file [Required]
+
+This file contains the license of the module. All the Ballerina Standard Library modules use the `Apache-2` license.
```
[the remark this answers is not in the document]
Comment (NipunaRanasinghe) [reply docs/module-contribution-guide.md]: Fixed with https://github.com/ballerina-platform/ballerina-extended-library/pull/505/commits/a735a8d72fe43bbbd39ad075d28ba5c396bb2b67

**Review by ThisaruGuruge — APPROVED** [review]
LGTM

**Pull request merged**